libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/QnAMakerRecognizer.java

```diff
@@ -0,0 +1,552 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.ai.qna;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Strings;
+import com.microsoft.bot.ai.qna.models.Metadata;
+import com.microsoft.bot.ai.qna.models.QnARequestContext;
+import com.microsoft.bot.ai.qna.models.QueryResult;
+import com.microsoft.bot.ai.qna.models.RankerTypes;
+import com.microsoft.bot.builder.IntentScore;
+import com.microsoft.bot.builder.RecognizerResult;
+import com.microsoft.bot.dialogs.DialogContext;
+import com.microsoft.bot.dialogs.ObjectPath;
+import com.microsoft.bot.dialogs.Recognizer;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.Serialization;
+
+/**
+ * IRecognizer implementation which uses QnAMaker KB to identify intents.
+ */
+public class QnAMakerRecognizer extends Recognizer {
+
+    @JsonProperty("$kind")
+    private final String kind = "Microsoft.QnAMakerRecognizer";
+
+    private final String qnAMatchIntent = "QnAMatch";
+
+    private final String intentPrefix = "intent=";
+
+    @JsonProperty("knowledgeBaseId")
+    private String knowledgeBaseId;
+
+    @JsonProperty("hostname")
+    private String hostName;
+
+    @JsonProperty("endpointKey")
+    private String endpointKey;
+
+    @JsonProperty("top")
+    private Integer top = 3;
+
+    @JsonProperty("threshold")
+    private Float threshold = 0.3f;
+
+    @JsonProperty("isTest")
+    private Boolean isTest;
+
+    @JsonProperty("rankerType")
+    private String rankerType = RankerTypes.DEFAULT_RANKER_TYPE;
+
+    @JsonProperty("strictFiltersJoinOperator")
+    private JoinOperator strictFiltersJoinOperator;
+
+    @JsonProperty("includeDialogNameInMetadata")
+    private Boolean includeDialogNameInMetadata = true;
+
+    @JsonProperty("metadata")
+    private Metadata[] metadata;
+
+    @JsonProperty("context")
+    private QnARequestContext context;
+
+    @JsonProperty("qnaId")
+    private Integer qnAId = 0;
+
+    @JsonProperty("logPersonalInformation")
+    private Boolean logPersonalInformation = false;
+
+    /**
+     * Initializes a new instance of the {@link QnAMakerRecognizer} class.
+     */
+    public QnAMakerRecognizer() {
```

Comment:
Why is this empty? if its being called won't it break the execution? and if it's not being called, shouldn't we delete it?

Reply:
Agree with you, I think that we can remove it.

I'll do it :)